connectors/api/gsheet.py

```
@@ -0,0 +1,137 @@
+        self.authentication_details = {AuthType.SERVICE_ACCOUNT: {
+            "spreadsheet_link": "",
+            "credentials_json": ""}
+        }
+        self.auth_url = "https://accounts.google.com/o/oauth2/auth"
+        self.token_url = "https://oauth2.googleapis.com/token"
+
+        self.main_response_key = "values"
+        self.required_libs = ["google-api-python-client", "google-auth-httplib2", "google-auth-oauthlib"]
+        super().__init__()
+
+    def connect_to_api(self, auth_type=AuthType.SERVICE_ACCOUNT, **auth_params) -> bool:
```

Ruff (0.8.2)

`AuthType` may be undefined, or defined from star imports

(F405)